yt_dlp/extractor/dorna.py

```diff
@@ -0,0 +1,393 @@
+                    for fmt in temp_formats:
+                        fmt.update({'language': lang})
+                formats.extend(temp_formats)
+
+        return {
+            'title': strip_or_none(traverse_obj(response, ('video_info', 'title_en'))),
+            'alt_title': strip_or_none(traverse_obj(response, ('video_info', 'title'))),
+            'thumbnail': traverse_obj(response, ('video_info', 'urlimage')),
+            'duration': int_or_none(traverse_obj(response, ('video_info', 'duration'))),
+            'formats': formats,
+            'chapters': self._chapters_from_url(
+                traverse_obj(response, ('video_info', 'clips_and_tags_url'))),
+        }
+
+
+class MotoGPIE(DornaIE):
+    # login only works reliably with cookies, usr:psw triggers re-captcha most of the time
+    _HOST = 'https://www.motogp.com'
+    _SECURE_HOST = 'https://secure.motogp.com'
+    _LOGIN_URL = f'{_SECURE_HOST}/en/user/login?return_to={_HOST}'
+    _VALID_URL = rf'''(?x)(?:
+                              motogp:({DornaIE._LANGS}):|
+                              (?:{_HOST}|{_SECURE_HOST})/({DornaIE._LANGS})/videos/.+/
+                          )(?P<id>\d{{5,}})'''
+    _TESTS = [{
+        # Free video: no account required
+        'url': 'https://www.motogp.com/it/videos/2022/12/01/motogp-and-f1-take-centre-stage-as-they-look-to-the-future/445783',
+        'info_dict': {
+            'id': '445783',
+            'ext': 'mp4',
+            'title': 'MotoGP™ and F1 take centre stage as they look to the future',
+            'alt_title': 'Dove porta il futuro di MotoGP™ e F1?',
+            'thumbnail': r're:^https://photos\.motogp\.com/.+jpg',
+            'upload_date': '20221201',
+            'timestamp': 1669905344,
+            'duration': 166,
+            'chapters': 'count:0',
+            'formats': 'count:14',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }, {
+        # Free account required: multi source video (broadcast, helicopter, on-board ...) with chapters
+        'url': 'https://www.motogp.com/it/videos/2019/08/11/round-11-motogp-myworld-motorrad-grand-prix-von-osterreich/304604',
+        'info_dict': {
+            'id': '304604',
+            'ext': 'mp4',
+            'title': 'Round 11: MotoGP™ myWorld Motorrad Grand Prix von Österreich',
+            'alt_title': 'Dal Red Bull Ring, la gara MotoGP™',
+            'thumbnail': r're:^https://photos\.motogp\.com/.+jpg',
+            'upload_date': '20190811',
+            'timestamp': 1565529508,
+            'duration': 5121,
+            'chapters': 'count:22',
+            'formats': 'count:83',
+        },
+        'params': {
+            'skip_download': True,
+        },
+        'skip': 'Account needed',
+    }]
+    _NETRC_MACHINE = 'motogp'
+
+    def _real_extract(self, url):
+        lang1, lang2, video_id = self._match_valid_url(url).groups()
+        self._LANG = lang1 or lang2
```

Comment:
Don't mutate class vars based on a URL. This indicates poor design of the methods

Reply:
unresolved